### ISSUE TITLE
@route directive in .dothml files

### ISSUE THREAD

**Issue by djanosik** [riganti/dotvvm#190]
It would be nice to have `@route` directive in .dothml files. It would make defining routes much more convenient.

```html
@viewModel Whatever.ViewModel, InSomeAssembly
@route AdminCustomer, "admin/customer/{Id:int}", { Id = 0 }

<dot:Content ContentPlaceHolderID="Body">
  ...
</dot:Content>
```

What do you think?

**Comment by exyi** [comment]
I suppose this could be quite nice, but there is a problem that .dothtml files are lazy loaded, based on their path in the route table. So you can't have route definitions in the .dothtml file, it would be chicked and egg problem :(. Maybe it could be solved by parsing all the files in folder based routing strategy, but it's not that easy as it might look like.

**Comment by djanosik** [comment]
I have a working [prototype](https://gist.github.com/djanosik/faa3dc705dc9235c01d24fe74c207a71) based on HTML comments. It would require some more sophisticated solution, but it should be possible.

**Comment by tomasherceg** [comment]
I don't think it is a problem to parse all files during application startup - the parser is quick (or it should be since we are using it heavily in the VS Extension) and we don't even have to parse the whole page, only the directives section. 

We can start with the implementation right after the 1.0 is RTM.

**Comment by tomasherceg** [comment]
Since we have implemented default values in the route URL (https://github.com/riganti/dotvvm/pull/680), the syntax of the `@route` directive could now be simplified:

```
@route admin/customer/{Id:int=1}
```

Also, we can probably generate the route name automatically, and allow it to be overridden by another directive:

```
@route admin/customer/{Id:int=1}
@routeName AdminCustomer
```
